src/utilities/caching/sizes.ts

```diff
@@ -0,0 +1,263 @@
+import { global } from "../globals/index.js";
+
+declare global {
+  interface Window {
+    [cacheSizeSymbol]?: Partial<CacheSizes>;
+  }
+}
+
+/**
+ * The cache sizes used by various Apollo Client caches.
+ *
+ * Note that these caches are all derivative and if an item is cache-collected,
+ * it's not the end of the world - the cached item will just be recalculated.
+ *
+ * As a result, these cache sizes should not be chosen to hold every value ever
+ * encountered, but rather to hold a reasonable number of values that can be
+ * assumed to be on the screen at any given time.
+ *
+ * We assume a "base value" of 1000 here, which is already very generous.
+ * In most applications, it will be very unlikely that 1000 different queries
+ * are on screen at the same time.
+ */
+export interface CacheSizes {
+  /**
+   * Cache size for the [`print`](../../utilities/graphql/print.ts) function.
+   *
+   * @defaultValue
+   * Defaults to `2000`.
+   *
+   * @remarks
+   * This method is called from the `QueryManager` and various `Link`s,
+   * always with the "serverQuery", so the server-facing part of a transformed
+   * DocumentNode.
+   */
+  print: number;
+  /**
+   * Cache size for the [`parser`](../../react/parser/index.ts) function.
+   *
+   * @defaultValue
+   * Defaults to `1000`.
+   *
+   * @remarks
+   * This function is used directly in HOCs, and nowadays mainly accessed by
+   * calling `verifyDocumentType` from various hooks.
+   * It is called with a user-provided DocumentNode.
+   */
+  parser: number;
+  /**
+   * Cache size for the `performWork` method of each [`DocumentTransform`](../../utilities/graphql/DocumentTransform.ts).
+   *
+   * @defaultValue
+   * Defaults to `2000`.
+   *
+   * @remarks
+   * This method is called from `transformDocument`, which is called from
+   * `QueryManager` with a user-provided DocumentNode.
+   * It is also called with already-transformed DocumentNodes, assuming the
+   * user provided additional transforms.
```

Comment:
```suggestion
   * This cache holds the results from transformed DocumentNodes to prevent
   * unnecessary recomputation. Document transforms are used whenever a query
   * is executed in the client.
```

Again, I would try and avoid some of the internal mechanics of where its called and instead provide a broader overview here. `DocumentTransform`s are also called in `ObservableQuery` as well, but I don't think we need to provide a reference to every location it could be used (and who knows, we may add to it in the future, so I'd hate to keep this comment up-to-date).

Reply:
It's difficult - here I provide this much details because this cache is **not** a DocumentTransform cache and should not be confused with those. 
It's also important to distinguish here the they are called with a user-provided DocumentNode (so they require a smaller cache size than DocumentTransforms) would.  
It's less important for cache size what the result is, and much more important what the input/cache key is going to be.